src/custom/pages/Claim/Investment/InvestOption.tsx

```diff
@@ -0,0 +1,157 @@
+
+  const vCowAmount = useMemo(() => {
+    if (!token || !price) {
+      return
+    }
+
+    const investA = CurrencyAmount.fromRawAmount(token, parseUnits(investedAmount, token.decimals).toString())
+    return investA.multiply(price)
+  }, [investedAmount, price, token])
+
+  return (
+    <InvestTokenGroup>
+      <div>
+        <span>
+          <TokenLogo symbol={currencyAmount?.currency?.symbol || '-'} size={72} />
+          <CowProtocolLogo size={72} />
+        </span>
+        <h3>Buy vCOW with {currencyAmount?.currency?.symbol}</h3>
+      </div>
+
+      <span>
+        <InvestSummary>
+          <span>
+            <b>Price</b>{' '}
+            <i>
+              {formatSmart(price)} vCoW per {currencyAmount?.currency?.symbol}
+            </i>
+          </span>
+          <span>
+            <b>Token approval</b>
+            <i>
+              {approveState === ApprovalState.NOT_APPROVED ? (
+                `${currencyAmount?.currency?.symbol} not approved`
+              ) : (
+                <Row>
+                  {currencyAmount?.currency?.symbol} approved{' '}
+                  <CheckCircle color="lightgreen" style={{ marginLeft: 5 }} />
+                </Row>
+              )}
+            </i>
+            {approveState === ApprovalState.NOT_APPROVED && (
+              <button onClick={approveCallback}>Approve {currencyAmount?.currency?.symbol}</button>
+            )}
+          </span>
+          <span>
+            <b>Max. investment available</b>{' '}
+            <i>
+              {formatSmart(maxCost) || '0'} {currencyAmount?.currency?.symbol}
+            </i>
+          </span>
+          <span>
+            <b>Available investment used</b>
+
+            <div>
+              <RangeSteps>
+                {[0, 25, 50, 75, 100].map((step: number) => (
+                  <RangeStep onClick={() => handleStepChange(step)} key={step}>
+                    {step}%
+                  </RangeStep>
+                ))}
+              </RangeSteps>
+
+              <input
+                style={{ width: '100%' }}
+                onChange={handlePercentChange}
+                type="range"
+                min="0"
+                max="100"
+                value={0}
+              />
+            </div>
+          </span>
+        </InvestSummary>
+        <InvestInput>
+          <div>
+            <span>
+              <b>Balance:</b>{' '}
+              <i>
+                {formatSmart(balance)} {currencyAmount?.currency?.symbol}
+              </i>
+              {/* Button should use the max possible amount the user can invest, considering their balance + max investment allowed */}
+              <button onClick={onMaxClick}>Invest max. possible</button>
+            </span>
+            <label>
+              <b>{currencyAmount?.currency?.symbol}</b>
+              <input disabled placeholder="0" value={investedAmount} max={formatSmart(currencyAmount)} />
```

Comment:
Actually the max might be different than `currencyAmout`. Maybe we could export the logic from `onMaxClick` to be re-used here